runtime/services/reflection.go

```diff
@@ -23,44 +16,11 @@ type ReflectionService struct {
 }
 
 func NewReflectionService() (*ReflectionService, error) {
-	fds := &descriptorpb.FileDescriptorSet{}
-
-	// load gogo proto file descriptors
-	allFds := proto.AllFileDescriptors()
-	haveFileDescriptor := map[string]bool{}
-	for _, compressedBz := range allFds {
-		rdr, err := gzip.NewReader(bytes.NewReader(compressedBz))
-		if err != nil {
-			return nil, err
-		}
-
-		bz, err := io.ReadAll(rdr)
-		if err != nil {
-			return nil, err
-		}
-
-		fd := &descriptorpb.FileDescriptorProto{}
-		err = protov2.Unmarshal(bz, fd)
-		if err != nil {
-			return nil, err
-		}
-
-		fds.File = append(fds.File, fd)
-		haveFileDescriptor[*fd.Name] = true
+	fds, err := reflection.GetFileDescriptorSet()
```

Comment:
Moved the actual logic to `"github.com/cosmos/cosmos-sdk/types/reflection"` to be re-usable (by baseapp)